spec/factories/course_condition_achievements.rb

```diff
@@ -3,5 +3,22 @@
           class: Course::Condition::Achievement.name, aliases: [:achievement_condition] do
     course
     achievement
+    association :conditional, factory: :course_achievement
+
+    factory :self_referential_course_condition_achievement do
+      after(:build) do |achievement_condition, evaluator|
+        achievement_condition.achievement = achievement_condition.conditional
+      end
+      to_create { |instance| instance.save(validate: false) }
+    end
+
+    factory :duplicate_child_course_condition_achievement do
+      after(:build) do |achievement_condition, evaluator|
```

Comment:
Unused block argument - `achievement_condition`. If it's necessary, use `_` or `_achievement_condition` as an argument name to indicate that it won't be used.